src/components/core/TextButton.tsx

```diff
@@ -0,0 +1,42 @@
+import { merge } from 'lodash'
+import { ButtonProps, Pressable, PressableProps, Text } from 'react-native'
+
+import { makeStyles } from 'app/styles'
+
+const useStyles = makeStyles(({ typography, palette }, { variant }) => {
+  const variantStyles = {
+    primary: {
+      text: {
+        color: palette.primary
+      }
+    },
+    secondary: {
+      text: {
+        color: palette.secondary
+      }
+    }
+  }
+
+  const baseStyles = {
+    text: {
+      ...typography.body
+    }
+  }
+
+  return merge(baseStyles, variantStyles[variant])
+})
+
+type TextButtonProps = PressableProps &
+  ButtonProps & {
+    variant: 'primary' | 'secondary'
```

Comment:
We use `type` for the button component, should they be consistent? If so, I'd vote for `variant` everywhere because type is so overloaded (and my webdev brain thinks of `<input type="submit">` etc)

Reply:
yes i also agree with variant, since type is overloaded, that was the intention here! adding a linear for moving everything to variant